File: server/domain/entities/prompt.py
"""Value Object - Prompt"""
from dataclasses import dataclass
@dataclass(frozen=True)
class Prompt:
    text: str
    negative_text: str = ""
    def __post_init__(self) -> None:
        if not self.text or not self.text.strip():
            raise ValueError("Prompt text cannot be empty")
        if len(self.text) > 1000:
            raise ValueError("Prompt text cannot exceed 1000 characters")
        object.__setattr__(self, "text", self.text.strip())

File: client/src/core/domain/prompt.ts
export interface Prompt {
  text: string;
  negativeText?: string;
}
export function createPrompt(text: string, negativeText?: string): Prompt {
  if (!text || text.trim().length === 0) {
    throw new Error("Prompt text cannot be empty");
  }
  if (text.length > 1000) {
    throw new Error("Prompt text cannot exceed 1000 characters");
  }
  return { text: text.trim(), negativeText: negativeText?.trim() };
}